frontend/src/pages/projects/screens/detail/notebooks/useNotebookImage.ts

```diff
@@ -2,35 +2,52 @@ import { NotebookKind } from '~/k8sTypes';
 import {
   getImageStreamDisplayName,
   getImageVersionDependencies,
-  getRelatedVersionDescription,
+  getImageVersionSoftwareString,
 } from '~/pages/projects/screens/spawner/spawnerUtils';
-import { ImageVersionDependencyType } from '~/pages/projects/screens/spawner/types';
 import useNotebookImageData from './useNotebookImageData';
-
-export type NotebookImage = {
-  imageName: string;
-  tagSoftware?: string;
-  dependencies: ImageVersionDependencyType[];
-};
+import { NotebookImageAvailability } from './const';
+import { NotebookImage } from './types';
 
 const useNotebookImage = (
   notebook: NotebookKind,
-): [notebookImage: NotebookImage | null, loaded: boolean] => {
-  const [imageData, loaded] = useNotebookImageData(notebook);
+):
+  | [notebookImage: null, loaded: false, loadError: undefined]
+  | [notebookImage: null, loaded: true, loadError: Error]
+  | [notebookImage: NotebookImage, loaded: true, loadError: undefined] => {
+  const [data, loaded, loadError] = useNotebookImageData(notebook);
+
+  if (!notebook || !loaded) {
+    return [null, false, undefined];
+  }
+  if (loadError) {
+    return [null, true, loadError];
```

Comment:
Similar comment elsewhere:
```suggestion
    return [null, false, loadError];
```